src/v/cluster/partition_probe.cc

```diff
@@ -59,6 +59,27 @@ void replicated_partition_probe::setup_internal_metrics(const model::ntp& ntp) {
       partition_label(ntp.tp.partition()),
     };
 
+    // The following few metrics uses a separate add_group call which doesn't
+    // aggregate any labels since aggregation does not make sense for "leader
+    // ID" values.
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("cluster:partition"),
+      {sm::make_gauge(
+         "leader_id",
+         [this] {
+             return _partition.raft()->get_leader_id().value_or(
+               model::node_id(-1));
+         },
+         sm::description("Id of current partition leader"),
+         labels),
+       sm::make_gauge(
+         "under_replicated_replicas",
+         [this] {
+             return _partition.raft()->get_under_replicated().value_or(0);
+         },
+         sm::description("Number of under replicated replicas"),
+         labels)});
```

Comment:
Here I think you still want to aggregate the shard label?

Reply:
Yeah, I guess. As we've discussed before aggregating the shard label when a metric is per-partition isn't very useful: it throws away the shard info without reducing the number of series but I guess any change there can be in a different series.

Reply:
Fixed.